src/charm.py

```diff
@@ -1199,21 +1188,20 @@ def _encode_openid_team_map(team_map: str) -> str:
             team_map (str): team definition.
 
         Returns:
-            A PHP array, as a Python string.
+            A Python structure that will be converted to json
         """
-        array_items = []
+        teams_parsed: dict = {}
         for idx, mapping in enumerate(team_map.split(","), start=1):
             launchpad_role, wordpress_role = mapping.split("=")
             launchpad_role = launchpad_role.strip()
             wordpress_role = wordpress_role.strip()
-            array_items.append(
-                f"{idx} => (object) array ("
-                f"'id'=>{idx},"
-                f"'team'=>'{launchpad_role}',"
-                f"'role'=>'{wordpress_role}',"
-                f"'server' => '0',),"
-            )
-        return f"array({''.join(array_items)})"
+            teams_parsed[str(idx)] = {
+                "id": idx,
+                "team": launchpad_role,
+                "role": wordpress_role,
+                "server": "0",
```

Comment:
Out of curiosity, what does the server represent? - (Optionally, would it be possible to have this in a typed dict with attributes so these can be explained within docstrings?)

Reply:
Looks like it is a way of mapping servers to [open id endpoints](https://git.launchpad.net/~canonical-sysadmins/wordpress-teams-integration/+git/wordpress-teams-integration/tree/openid-teams.php?id=a643325df2e40ad5a34346f93872c47cdaa09051#n50) and 0 is the only value used... To be honest, it looks like something not used...

I will merge as I do not plan to work more in Wordpress charm, but I believe the correct action, beside documenting as you state, would be to work on the plugins...